packages/slice-machine/src/domain/customType.ts

```diff
@@ -90,13 +90,41 @@ export function getMainSectionEntry(
   return sections[0];
 }
 
+export function getMainSectionId(customType: CustomType): string | undefined {
+  return Object.keys(customType.json)[0];
+}
+
 export function getSection(
   customType: CustomType,
   sectionId: string,
 ): DynamicSection | undefined {
   return customType.json[sectionId];
 }
 
+export function getSectionWithUIDField(
+  customType: CustomType,
+): [string, DynamicSection] | undefined {
+  const sections = getSectionEntries(customType);
+  const sectionWithUID = sections.find(([_, section]) =>
+    Object.values(section).some((field): field is UID => field.type === "UID"),
+  );
+  return sectionWithUID;
+}
+
+export function getUIDField(customType: CustomType): UID | undefined {
+  const sectionEntry = getSectionWithUIDField(customType);
+  const [_, section] = sectionEntry ?? [];
+
+  return Object.values(section ?? {}).find(
+    (field): field is UID => field.type === "UID",
+  );
+}
+
+export function getUIDFieldLabel(customType: CustomType): string | undefined {
```

Comment:
💡 #idea

Let's be more generic here, we just want a way to get the label from a field as it's always the same way to get it.

```suggestion
export function getFieldLabel(field: NestableWidget | UID | Group): string | undefined {
```

Reply:
good idea, will change that